tests/Avalonia.Controls.UnitTests/ButtonTests.cs

```diff
@@ -92,6 +95,164 @@ public void Button_Is_Disabled_When_Disabled_Bound_Command_Is_Added()
             Assert.False(target.IsEnabled);
         }
 
+        [Fact]
+        public void Button_Is_Raising_Click()
+        {
+            var mouse = Mock.Of<IMouseDevice>();
+            IInputElement captured = null;
+            Mock.Get(mouse).Setup(m => m.GetPosition(It.IsAny<IVisual>())).Returns(new Point(50, 50));
+            Mock.Get(mouse).Setup(m => m.Capture(It.IsAny<IInputElement>())).Callback<IInputElement>(v => captured = v);
+            Mock.Get(mouse).Setup(m => m.Captured).Returns(() => captured);
+
+            var target = new TestButton() { Bounds = new Rect(0, 0, 100, 100) };
+
+            bool clicked = false;
+
+            target.Click += (s, e) => clicked = true;
+
+            RaisePointerEnter(target, mouse);
+            RaisePointerMove(target, mouse);
+            RaisePointerPressed(target, mouse, 1, MouseButton.Left);
+
+            Assert.Equal(captured, target);
+
+            RaisePointerReleased(target, mouse, MouseButton.Left);
+
+            Assert.Equal(captured, null);
+
+            Assert.True(clicked);
+        }
+
+        [Fact]
+        public void Button_Is_Not_Raising_Click_When_PointerReleased_Outside()
```

Comment:
Nit: should be called "Button_Does_Not_Raise_Click_When_PointerReleased_Outside"